Refactor measures

annapowellsmith opened the pull request:
A couple of typo fixes, plus more significant improvements:

1. Show only standard practices on measures pages. We only use standard practices to calculate our CCG-level measures, and therefore we should only show these practices on CCG pages.

2. Make this code less of a giant hairball of doom. Create a separate file for data logic. Add tests for this file. Keep only presentation logic in the original file. This changes nothing visible on the front-end, but makes the code much more maintainable, and will be a good basis for using a framework in the future.

Fixes #66 among other things.

sebbacon commented on openprescribing/media/js/measures.js:
```diff
@@ -1,541 +1,161 @@
+    Highcharts.setOptions({
+        global: { useUTC: false }
+    });
+    L.mapbox.accessToken = 'pk.eyJ1IjoiYW5uYXBvd2VsbHNtaXRoIiwiYSI6ImNzY1VpYkkifQ.LC_IcHpHfOvWOQCuo5t7Hw';
```
Does it matter that our access token is public?

annapowellsmith replied:
I'm not sure there's much we can do about it: https://www.mapbox.com/help/rotate-access-token/

sebbacon commented on openprescribing/media/js/measures.js:
```diff
@@ -1,541 +1,161 @@
-    },
+                })
+                .fail(function(){
+                    console.log('data failed');
```
If we're bothering to log something to the console, perhaps log details of the error? E.g. something like:

    .fail(function (jqXHR, textStatus, error) {
        console.log("Error " + error + " when making request " + jqXHR);
    });

sebbacon commented on openprescribing/media/js/test/test_measures.js:
```diff
@@ -0,0 +1,743 @@
+    });
+
+    describe('#_getChartTitleEtc', function () {
+        it('should get explanation for non-cost-based measures', function () {
```
I don't understand this test (or its description): isn't this a cost-based measure?

annapowellsmith replied:
typo I think! and i need to add a test for non-cost-base ones. thanks.

sebbacon commented:
Overall, looks great. Appreciate the javascript function documentation. 

Inline comments all added now.

As a general point, I think you need to set your javascript indent to 2 spaces in your text editor to meet our linting standards.  Run `eslint <name of file>` to get a list of warnings and errors; they are mostly about the indent!

annapowellsmith commented:
Thanks, fixed inline comments. Also added a new commit with some descriptions in the JSON files. 

CodeClimate is failing but not giving details of the error. Suggestions? I have updated `src/measure_utils.js` to run cleanly through eslint - I think that's the only one that gets checked. Or is it all JS files, not just the ones in `src`?

sebbacon commented on openprescribing/frontend/management/commands/measure_definitions/bens.json:
```diff
@@ -193,10 +188,11 @@
         ],
         "why_it_matters": [
-            "TBA"
+            "There are a number of angiotensin receptor blockers (ARBs) available. A number of them are now available generically, and are of low cost. NICE guidance for treating high blood pressure recommends that the low-cost versions should be used."
```
"a number of" repeated. Maybe "There are several angiotensin..."?

sebbacon commented on openprescribing/frontend/management/commands/measure_definitions/bens.json:
```diff
@@ -273,10 +270,11 @@
         ],
         "why_it_matters": [
-            "TBA"
+            "It is recommended that when treating high cholesterol doctors prescribe statins that both reduce cholesterol by at least 40%, and that are low-cost.  The low-cost statins (simvastatin and atorvastatin) are suitable for the majority of patients."
```
Hard to parse. Perhaps:

"When treating high cholesterol, doctors are recommended to prescribe statins which are both low-cost, and reduce cholesterol by at least 40%" ?

sebbacon commented on openprescribing/frontend/management/commands/measure_definitions/ktts.json:
```diff
@@ -8,7 +8,7 @@
         ],
         "why_it_matters": [
-            "TBA"
+            "Latest NICE guidance suggest that, for the vast majority of people who need treatment for high cholesterol, these drugs should not be used.  There is no evidence that omega 3 fatty acids reduces the chances of patients getting cardiovascular disease."
```
Latest NICE guidance suggest**s** that

sebbacon commented:
As discussed, it's currently only the JS files in `src/` that are checked (though this should be changed at some point). 

For some reason the codeclimate analysis didn't happen. I suggest not worrying about that for now and I'll monitor the situation and email their support team over the next week or two as we do more PRs.

I'm happy for this branch to be merged once the wording stuff is addressed!

P.S. I'm thinking it would have been better to keep the lint-based changes in their own commit, so I could easily see what had changed since my last review (e.g. the addition of a new test). What do you think?

annapowellsmith commented:
Top proofing, thanks!

Yes, sorry, the linting should have been a separate commit! Will do it first from now on.

annapowellsmith commented:
Just added a couple of minor changes - some placeholder text, and a typo fix.
